src/ray/raylet/scheduling/policy/bundle_scheduling_policy.h

```diff
@@ -0,0 +1,127 @@
+// Copyright 2021 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <vector>
+
+#include "ray/raylet/scheduling/policy/scheduling_context.h"
+#include "ray/raylet/scheduling/policy/scheduling_policy.h"
+#include "ray/raylet/scheduling/policy/scorer.h"
+
+namespace ray {
+namespace raylet_scheduling_policy {
+
+class BundleSchedulingPolicy : public ISchedulingPolicy {
+ public:
+  explicit BundleSchedulingPolicy(
+      const absl::flat_hash_map<scheduling::NodeID, Node> &nodes,
+      std::function<bool(scheduling::NodeID)> is_node_available,
+      std::function<bool(scheduling::NodeID, const ResourceRequest &)>
```

Comment:
this is a bit unfortunate. should the scheduling policy take a ClusterResourceManager?

Reply:
> this is a bit unfortunate. should the scheduling policy take a ClusterResourceManager?

Ok, I will fix it soon.